### PR TITLE
DF-1110

### PR DESCRIPTION
**Pull request by dvoracek-martin**

### CODE REVIEW

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChannelDto.java [hunk 1]
```diff
@@ -65,4 +69,17 @@ public ApiApplyChannelDto setAdditionalInfo(String additionalInfo) {
         this.additionalInfo = additionalInfo;
         return this;
     }
+
+    @AssertTrue
+    private boolean isEitherEmailOrMail() {
```
Comment (birom) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChannelDto.java]: The JobAdvertisementFromWebAssembler.validateApplyChannelIsNotEmpty has a different logic please check with @cjpsycos how should this look like. 
Please also have a look at the ``ValidWorkingTimeRange`` and the ``ValidWorkingTimeRangeValidator``. I think doing cross field validation like that is a better approach.

Comment (fabiangotzen) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChannelDto.java]: As far as I know the logic for the ApiApplyChannelDto should be the same in this case. So a Validator with 
f.E
```
    public boolean isValid(ApiApplyChannelDto apiApplyChannelDto, ConstraintValidatorContext context) {
        return  (apiApplyChannelDto.getMailAddress() != null)
                || hasText(apiApplyChannelDto.getEmailAddress())
                || hasText(apiApplyChannelDto.getPhoneNumber())
                || hasText(apiApplyChannelDto.getFormUrl());
    }
```
should work in this case.

Comment (dvoracek-martin) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChannelDto.java]: adjusted

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChannelDto.java [hunk 2]
```diff
@@ -4,21 +4,22 @@
 
 import javax.validation.constraints.Size;
 
+@ApiApplyChanelCustomConstraint
```
Comment (birom) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChannelDto.java]: I think a name like "AtleastOneApplyChannelRequired" would be better.

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChanelCustomConstraint.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = ApiApplyChanelCustomConstraintValidator.class)
+@Documented
+public @interface ApiApplyChanelCustomConstraint {
```
Comment (cjpsycos) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChanelCustomConstraint.java]: Typo Channel

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChanelCustomConstraintValidator.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package ch.admin.seco.jobs.services.jobadservice.infrastructure.web.controller.api;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static org.springframework.util.StringUtils.hasText;
+
+public class ApiApplyChanelCustomConstraintValidator implements ConstraintValidator<ApiApplyChanelCustomConstraint, ApiApplyChannelDto> {
```
Comment (cjpsycos) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/ApiApplyChanelCustomConstraintValidator.java]: Typo Channel

**Pull request merged**